### DISCUSSION TITLE
Release 3.5.0 - CPU Fan manual selection & bugfix

### DISCUSSION THREAD

**Discussion by mathoudebine** [mathoudebine/turing-smart-screen-python#519] [Announcements]
## New feature : CPU fan manual selection for Linux platforms
On Linux platforms, the CPU fan is often controlled by the motherboard unlike the GPU fan.
Because of that, it does not contain `cpu` in the sensor name, and cannot automatically detected.
E.g. with an AMD CPU on an Asus motherboard: the CPU fan sensor is managed by `nct6798` Nuvoton's Super I/O Controller
```sh
>$ sensors
nct6798-isa-0290
Adapter: ISA adapter
fan1:                      707 RPM  (min =    0 RPM)        #
fan2:                      906 RPM  (min =    0 RPM)        #   One of those is the CPU fan, but which one?
fan3:                      691 RPM  (min =    0 RPM)        #
fan4:                      657 RPM  (min =    0 RPM)        #
fan5:                        0 RPM  (min =    0 RPM)
fan6:                        0 RPM  (min =    0 RPM)
fan7:                        0 RPM  (min =    0 RPM)

amdgpu-pci-0800
Adapter: PCI adapter 
fan1:         825 RPM  (min =    0 RPM, max = 3200 RPM)     #   GPU fan is clearly identified
```

On previous releases, when CPU fan could not be automatically detected, it was ignored with the following warning on logs:
```
20/02/2024 18:05:13 [WARNING] Your CPU Fan Speed is not supported yet
```

With this new release, there is now a new field in the `config.yaml` configuration file:
```yaml
  # CPU fan
  # For Linux/MacOS platforms, the CPU fan is amongst all fan sensors gathered from the motherboard chipset
  # If value is AUTO the system monitor will try to auto-select the CPU fan
  # If auto-detection fails, it might be necessary to manually indicate which fan is the CPU fan
  # Value must be 'controller/fan' e.g. 'nct6798/fan2'. Use configuration wizard for help in selection
  CPU_FAN: AUTO
```

And the CPU fan can be manually selected in the configuration wizard:

![image](https://github.com/mathoudebine/turing-smart-screen-python/assets/38615348/162e8fed-2241-445c-8a49-d5136b9460d9)

A tooltip also indicates which package to install in order to detect hardware sensors from the motherboard:

![image](https://github.com/mathoudebine/turing-smart-screen-python/assets/38615348/1a389747-ee43-44b5-83d7-9703416dc0b7)

## What's Changed
* Fix sensor names for Intel GPU, avoid exception when fan speed is None by @mathoudebine in https://github.com/mathoudebine/turing-smart-screen-python/pull/479
* Add GPU frequency sensor by @mathoudebine in https://github.com/mathoudebine/turing-smart-screen-python/pull/480
* :arrow_up: Update pillow requirement from ~=10.2.0 to ~=10.3.0 by @dependabot in https://github.com/mathoudebine/turing-smart-screen-python/pull/496
* Add LHM robustness for sensors None values by @mathoudebine in https://github.com/mathoudebine/turing-smart-screen-python/pull/507
* Check if theme is valid for Configuration Wizard by @mathoudebine in https://github.com/mathoudebine/turing-smart-screen-python/pull/506
* :arrow_up: Update babel requirement from ~=2.14.0 to ~=2.15.0 by @dependabot in https://github.com/mathoudebine/turing-smart-screen-python/pull/511
* Add CPU fan selection for Python libraries on Linux by @mathoudebine in https://github.com/mathoudebine/turing-smart-screen-python/pull/484

**Full Changelog**: https://github.com/mathoudebine/turing-smart-screen-python/compare/3.4.0...3.4.1

<hr /><em>This discussion was created from the release <a href='https://github.com/mathoudebine/turing-smart-screen-python/releases/tag/3.5.0'>Release 3.5.0 - CPU Fan manual selection & bugfix</a>.</em>

**Comment by Jordi-print** [comment]
Una consulta, no soy conocer del tema, pero donde esta el software?

**Reply by DJRoby19** [reply]
https://github.com/mathoudebine/turing-smart-screen-python.git

**Comment by DJRoby19** [comment]
Great work in version 3.5.0 only problem on cyberdeck theme everything works except the GPU fan but in the drop down menu it is visible and working but on the theme no instead on version 3.4.0 everything worked except the CPU fan how could I solve it?

**Comment by seythegamer** [comment]
Thanks alot!

Ive been having some fun making my own themes using your app, but there is an issue with GPU.
Im using AMD 7800 XT, and none of the graphs shows anything.

For an example: CPU 'new' line graph works properly - shows history for whatever stat i want it to show, but theres nothing on gpu.
In the CMD i can see that my gpu has been recognized, but unfortunately only temperature TEXT works (Can't get the graph to show it) (also % usage of gpu updates poorly or is stuck on same number - only temperature TEXT works).

When can we expect an update? Or is there any easy fix for non-it guy like me?

Also it would be really great if there was possibility to place line graph one on another (sample further down)
For an example (as you can see on my photo of the theme) now i am using two seperate places for networking info (downloading and uploading) it would be really cool if i could place them in one box and they would be drawn as shown on the picture \/
![taskmanager - gpu doesnt work](https://github.com/user-attachments/assets/8d838e38-7b67-49ac-bfde-fd34453d46ca)
![graphs](https://github.com/user-attachments/assets/e0d06ca1-3e58-4826-8a98-b2798cbe9686)





************EDIT1:
I figured out why my GPU graph wasn't drawing properly
i just had it TABbed wrongly
lol

**Reply by seythegamer** [reply]
Also is there an option to get disk usage info from each drive i got in my computer?
I can see that only showing the C: drive works for now.

Sincerely,
s

**Reply by c0ldJS** [reply]
This topic comes up semi-regularly. Might be interested in discussions #188 and #551.

**Reply by seythegamer** [reply]
thanks you so much,
will look into it in free time

im not used to python so it will take me a while to understand it
